src/zenml/models/v2/core/step_run.py

```diff
@@ -543,8 +550,42 @@ class StepRunFilter(WorkspaceScopedFilter):
     )
     model_version_id: Optional[Union[UUID, str]] = Field(
         default=None,
-        description="Model version associated with the pipeline run.",
+        description="Model version associated with the step run.",
         union_mode="left_to_right",
     )
+    model: Optional[Union[UUID, str]] = Field(
```

Comment:
I would propose renaming this attribute to something like `model_name_or_id`. I think it much more intuitive that way and also, `model` is a protected namespace by Pydantic. We are already bypassing that for some `model_XXX` attributes by using a custom `ConfigDict` but this one feels like the craziest 😄

Reply:
I think the UX with this renaming would be considerably worse: Instead of doing `zenml pipeline runs list --model=<MODEL_NAME> --user=<USER_ID>`, we would have `zenml pipeline runs list --model_name_or_id=<MODEL_NAME> --user_name_or_email_or_id=<USER_NAME>`, which is why we opted for the short names here. In my opinion, ideally these CLI options would be decoupled from the filter models, but right now that is not the case.
We have these shortcut names (including `model` on other filters already, so for consistency I'd keep them like this, is that okay?


(Regarding pydantic, I think `BaseModel.model` is actually more unlikely to be used by them compared to some other fields we have :P)

Reply:
> (Regarding pydantic, I think BaseModel.model is actually more unlikely to be used by them compared to some other fields we have :P)

I am very curious about how this will turn out in the future. 😄